### PR TITLE
Add the ability to add Firebase resources to an existing GCP project

### PR DESCRIPTION
**Pull request by TrCaM**
### Description
Design proposal: [go/firebase-cli-import](https://goto.google.com/firebase-cli-import)

- Added `firebase projects:addfirebase` command to add Firebase resources to an existing Google Cloud Platform (GCP) project
- Updated `firebase init` command with a new option to add Firebase to a GCP project and set it as the default project for the current directory

### Notes
- This PR has to be  merged to master after https://github.com/firebase/firebase-tools/pull/1562

### Testing
- Unit testing
- Manual testing

### CODE REVIEW

File: changelog.txt [hunk 1]
```diff
@@ -1,3 +1,3 @@
 * Add new project management commands: `projects:create`, `projects:list`
-* Add new app management commands: `apps:create`, `apps:list`, `apps:sdkconfig`
+* Add new app management commands: `apps:create`, `apps:list`, `apps:sdkconfig`, `apps:addfirebase
```
Comment (bkendall) [inline changelog.txt]: Missing a backtick

Comment (TrCaM) [reply changelog.txt]: Fixed

File: src/commands/projects-addfirebase.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import * as Command from "../command";
+import { FirebaseError } from "../error";
+import { addFirebaseToCloudProjectAndLog, promptAvailableProjectId } from "../management/projects";
+import * as requireAuth from "../requireAuth";
+
+module.exports = new Command("projects:addfirebase [projectId]")
+  .description("add Firebase resources to a Google Cloud Platform project")
+  .before(requireAuth)
+  .action(
+    async (projectId: string | undefined, options: any): Promise<any> => {
```
Comment (bkendall) [inline src/commands/projects-addfirebase.ts]: This returns `Promise<FirebaseProjectMetadata>`, doesn't it?

File: src/management/projects.ts [hunk 3]
```diff
@@ -256,10 +346,34 @@ export async function getProjectPage(
...
+    });
+  } catch (err) {
+    logger.debug(err.message);
+    throw new FirebaseError(
+      "Failed to list available Google Cloud Platform projects. See firebase-debug.log for more info.",
+      { exit: 2, original: err }
+    );
+  }
+
+  return projectPage;
```
Comment (bkendall) [inline src/management/projects.ts]: The value above (`await getProjectPage`) can be returned - the variable `projectPage` here is unnecessary.

File: src/commands/projects-addfirebase.ts (hunk 2 above)
Comment (TrCaM) [reply src/commands/projects-addfirebase.ts]: Updated other places as well

File: src/management/projects.ts (hunk 3 above)
Comment (TrCaM) [reply src/management/projects.ts]: Done

**Pull request merged**